src/lib/common/topology_sampler.cpp

```diff
@@ -64,53 +88,102 @@ static void topology_sample_cb(networkTopology_t* networkTopology) {
 
     memset(_node_list.nodes, 0, sizeof(_node_list.nodes));
 
     neighborTableEntry_t *cursor = NULL;
     uint16_t counter;
-    for(cursor = networkTopology->front, counter = 0; (cursor != NULL) && (counter < _node_list.num_nodes); cursor = cursor->nextNode, counter++) {
+    xQueueReset(_sys_info_queue);
+    xQueueReset(_config_cbor_map_queue);
+    bool exit = false;
+
+    // Iterate through the entire list of nodes and request the sys info.
+    for (cursor = networkTopology->front, counter = 0;
+         (cursor != NULL) && (counter < _node_list.num_nodes);
+         cursor = cursor->nextNode, counter++) {
       _node_list.nodes[counter] = cursor->neighbor_table_reply->node_id;
-      network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&_node_list.nodes[counter]), sizeof(uint64_t));
+      if (!sys_info_service_request(_node_list.nodes[counter], sys_info_reply_cb,
+                                    NODE_SYS_INFO_REQUEST_TIMEOUT_S)) {
+        printf("Failed to send sys info request to node: %" PRIu64 "\n",
+               _node_list.nodes[counter]);
+        exit = true;
+        break;
+      }
+    }
+    if (exit) {
+      break;
     }
 
-    bm_common_config_crc_t config_crc = {
-      .partition = BM_COMMON_CFG_PARTITION_SYSTEM,
-      .crc32 = _sys_cfg->getCborEncodedConfigurationCrc32(),
-    };
-
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&config_crc), sizeof(bm_common_config_crc_t));
-
-    bm_common_fw_version_t fw_info = {
-      .major = 0,
-      .minor = 0,
-      .revision = 0,
-      .gitSHA = 0,
-    };
-
-    getFWVersion(&fw_info.major, &fw_info.minor, &fw_info.revision);
-    fw_info.gitSHA = getGitSHA();
+    // Create the network info cbor array and calculate the crc32
+    size_t cbor_bufsize =
+        _node_list.num_nodes * (sizeof(SysInfoSvcReplyMsg::Data) +
+                                sizeof(ConfigCborMapSrvReplyMsg::Data) + NODE_CONFIG_PADDING);
+    uint8_t *cbor_buffer = static_cast<uint8_t *>(pvPortMalloc(cbor_bufsize));
+    configASSERT(cbor_buffer);
+    bool encoding_success = false;
+    do {
+      if (!create_network_info_cbor_array(cbor_buffer, cbor_bufsize)) {
+        printf("Failed to create network info cbor map\n");
+        break;
+      }
+      network_crc32_calc = crc32_ieee(cbor_buffer, cbor_bufsize);
+      encoding_success = true;
+    } while (0);
 
-    network_crc32_calc = crc32_ieee_update(network_crc32_calc, reinterpret_cast<uint8_t *>(&fw_info), sizeof(bm_common_fw_version_t));
+    if (cbor_buffer) {
+      // TODO: Printing the buffer for now, but we should send this to spotter before freeing. 
+      for(uint32_t i = 0; i < cbor_bufsize; i++) {
+        printf("%02x ", cbor_buffer[i]);
+        if(i % 16 == 15) { // print a newline every 16 bytes (for print pretty-ness)
+          printf("\n");
+        }
+      }
+      printf("\n");
+      vPortFree(cbor_buffer);
+    }
```

Comment:
I think the `vPortFree(cbor_buffer)` needs to move down around line 193, outside the top-level do-while.

Also, the printing/sending currently happens even if `encoding_success == false`. It needs to move either into the above nested do-while, around line 133, or somewhere else down below.